Source/SwiftLintFramework/Rules/InlineCommentRule.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  InlineCommentRule.swift
+//  SwiftLint
+//
+//  Created by Kim de Vos on 05/11/17.
+//  Copyright © 2017 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+private let nonSpace = "[^\\s]"
+private let noNewLine = "(?!\\n)"
+private let nonOrOneSpace = "\\s{0,1}"
+private let twoSpace = "\\s{2}"
+private let twoOrMoreSpace = "\\s{2,}"
+private let threeOrMoreSpace = "\\s{3,}"
+private let comment = "//"
+private let endOfStatement = "[{}()?,:;]++\(noNewLine)"
+
+public struct InlineCommentRule: ConfigurationProviderRule, OptInRule {
+
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "inline_comment",
+        name: "InlineComment",
+        description: "Inline comments should be in valid format.",
+        kind: RuleKind.lint,
+        nonTriggeringExamples: [
+            "// Good\nfunc foo() {\n}",
+            "func foo() {  // Good\n}",
+            "class Foo {var foo = Date()  // Good\n}",
+            "class Foo {var foo: Date?  // Good\n}",
+            "class Foo {var foo = [1,  // Good\n2]\n}"
+        ],
+        triggeringExamples: [
+        "func foo() ↓{  //Wrong\n}",
```

Comment:
these violation offsets are really weird. IMO, they should be before the start of a comment (e.g. `↓//`) or where the "error" really is (in this case, I think it's because there's no space between `//` and `Wrong`, so the marker could be `//↓Wrong`).